### PR TITLE
Bi-directional dshot for revo-mini

### PR DESCRIPTION
**Pull request by anbello**
Tested on Copter with BLHeli_S ESC with JazzMaverick FW 16.73.
DShot300 and DShot600 works good with ESC telemetry used for harmonic notch.
Thanks to @andyp1per

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/revo-mini-bdshot/hwdef.dat [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Bi-directional dshot version of omnibusf4pro
```
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/revo-mini-bdshot/hwdef.dat:1]: You probably want to adjust this comment

**Review by andyp1per — APPROVED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/revo-mini-bdshot/hwdef.dat (hunk 1 above)
Comment (anbello) [reply libraries/AP_HAL_ChibiOS/hwdef/revo-mini-bdshot/hwdef.dat]: OK thanks.

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
need to investigate CI issue

**Comment by anbello** [comment]
CI issue is due to this error:
```
../../ArduPlane/tuning.cpp: In member function 'virtual AP_Float* AP_Tuning_Plane::get_param_pointer(uint8_t)':
../../ArduPlane/tuning.cpp:158:46: error: 'class AC_PosControl' has no member named 'get_vel_z_p'; did you mean 'get_vel_z_pid'?
         return &plane.quadplane.pos_control->get_vel_z_p().kP();
```
It seems like CI don't consider this commit:
https://github.com/ArduPilot/ardupilot/commit/ff8b69fbad23a464231a6888a50490ec49be17fa#diff-f44cd4c7aa8408fc5e9f856706666e9b41b0f2fe0cf5ccd5fda1b36a4a1d88e5

But I don't know how CI works, sorry.

**Comment by andyp1per** [comment]
You could try a rebase, maybe it will go away

**Comment by anbello** [comment]
Thanks, I just see that the master used by me for this PR don't has that commit, so I have to rebase and re-push.

**Comment by tridge** [comment]
is this still needed now that #17109 is merged?

**Comment by anbello** [comment]
> is this still needed now that #17109 is merged?

Yes, this is the bdshot conf for for revo-mini "original" the one w/o ext I2C and RCin on USART6RX (as described on wiki).

In #17109 we added revo-mini-i2c with ext I2C and RCin on PB14 and respective bdshot conf (revo-mini-i2c-bdshot).

Maybe I have to modify wiki  to specify that now we have two variant of revo-mini each w or w/o bdshot (so four).

Thanks

**Review by tridge — APPROVED** [review]

**Pull request merged**